examples/counter-contract/src/lib.rs

```diff
@@ -1,36 +1,117 @@
 // Do not link against libstd (i.e. anything defined in `std::`)
-#![no_std]
-#![feature(alloc_error_handler)]
+#![cfg_attr(not(test), no_std)]
 
 // However, we could still use some standard library types while
 // remaining no-std compatible, if we uncommented the following lines:
 //
 // extern crate alloc;
 
-use miden::{Felt, StorageMap, StorageMapAccess, Word, component, felt};
+use miden_test_harness_macros::miden_test_block;
 
-/// Main contract structure for the counter example.
-#[component]
-struct CounterContract {
-    /// Storage map holding the counter value.
-    #[storage(slot(0), description = "counter contract storage map")]
-    count_map: StorageMap,
+#[cfg(not(test))]
```

Comment:
Instead of gating this behind `#[cfg(not(test))]`, it would make more sense to gate this behind `#[cfg(target_family = "wasm")]` - our test suite will never be built for the Wasm target, and even if it _was_, then the compilation error you ran into wouldn't be an issue anyway.